libshortfin/CMakeLists.txt

```diff
@@ -46,9 +44,98 @@ list(APPEND CMAKE_MODULE_PATH
 include(shortfin_library)
 
 # Dependencies.
-find_package(spdlog)
-find_package(xtensor)
-find_package(IREERuntime)
+
+if(SHORTFIN_BUNDLE_DEPS)
+  ## fmt
+  FetchContent_Declare(
+    fmt
+    GIT_REPOSITORY https://github.com/fmtlib/fmt.git
+    GIT_TAG        e69e5f977d458f2650bb346dadf2ad30c5320281 # 10.2.1 (sync with spdlog)
+  )
+
+  ## spdlog
+  # We build fmt from source instead, because we also use fmt.
+  set(SPDLOG_FMT_EXTERNAL ON)
+  FetchContent_Declare(
+    spdlog
+    GIT_REPOSITORY https://github.com/gabime/spdlog.git
+    GIT_TAG        2d4acf8cc321d7783d8f2e22e17a794c6d0e9450 # v1.14.1
+  )
+
+  ## xtl: required for xtensor
+  FetchContent_Declare(
+    xtl
+    GIT_REPOSITORY https://github.com/xtensor-stack/xtl.git
+    GIT_TAG        a7c1c5444dfc57f76620391af4c94785ff82c8d6 # v0.7.7
+  )
+
+  ## xtensor
+  FetchContent_Declare(
+    xtensor
+    GIT_REPOSITORY https://github.com/xtensor-stack/xtensor.git
+    GIT_TAG        3634f2ded19e0cf38208c8b86cea9e1d7c8e397d # v0.25.0
+  )
+
+  FetchContent_MakeAvailable(fmt spdlog xtl xtensor)
+else()
+  find_package(spdlog)
+  find_package(xtensor)
+endif()
+
+## iree runtime
+
+if (NOT SHORTFIN_IREE_SOURCE_DIR AND SHORTFIN_BUNDLE_DEPS)
+  FetchContent_Declare(
+    iree
+    GIT_REPOSITORY https://github.com/iree-org/iree.git
+    GIT_TAG candidate-20240821.992
+    # TODO: We shouldn't have to pull googletest when we are not building tests.
+    #       This needs to be fixed with IREE.
+    GIT_SUBMODULES "third_party/benchmark third_party/cpuinfo third_party/flatcc third_party/hip-build-deps third_party/googletest"
+    GIT_SHALLOW ON
+  )
+  FetchContent_GetProperties(iree)
+  if(NOT iree_POPULATED)
+    FetchContent_Populate(iree)
```

Comment:
This gives me
> CMake Warning (dev) at /usr/share/cmake-3.30/Modules/FetchContent.cmake:1953 (message):
  Calling FetchContent_Populate(iree) is deprecated, call
  FetchContent_MakeAvailable(iree) instead.  Policy CMP0169 can be set to OLD
  to allow FetchContent_Populate(iree) to be called directly for now, but the
  ability to call it with declared details will be removed completely in a
  future version.
Call Stack (most recent call first):
  CMakeLists.txt:99 (FetchContent_Populate)
This warning is for project developers.  Use -Wno-dev to suppress it.

Fine if addresses in a follow-up though.

Reply:
Yeah, that's a weird thing about FetchContent. They recommend using FetchContent_MakeAvailable, which already does a add_subdirectory. I like fetching and setting the directory directly. Can change it later if needed.